elastic/kibana
# [Event log] Removing outdated event log indices from being managed by ILM

Resolves https://github.com/elastic/kibana/issues/97755

## Summary

Currently, the event log bootstrapping code creates a single ILM policy `kibana-event-log-policy` and then an index template for every version number (`kibana-event-log-7.12.0-template`, `kibana-event-log-7.13.1-template`, etc). The ILM policy is linked to every index template, so when a Kibana instance gets upgraded, the older kibana event log indices continue to be rotated following the ILM policy.

This PR looks for indices that are linked to the ILM policy that are not the current version and removes them from the policy. This means the indices will still exist, but will no longer be managed by ILM. 

## To Verify
* change the ILM policy for the event log to roll over fast (`max_age: '5m'` is what I used) 
* start up a previous version of ES and Kibana (like 7.13) using `yarn es snapshot --ssl -E path.data=../data` in order to keep using the same data for testing
* verify that:
  * the event log policy was created - `kibana-event-log-policy`
  * an index template was created - `.kibana-event-log-7.13.3-template`
  * there is an initial index for 7.13 - `.kibana-event-log-7.13.3-000001`
  * the initial index is linked to the event log policy
* the poll interval for ILM is 10 minutes, so wait 10-ish minutes and verify that your event log index rolled over
  * there should now be two indices `.kibana-event-log-7.13.3-000001` and `.kibana-event-log-7.13.3-000002`
  * both these indices should be linked to the event log policy
* switch to this branch and start up using `yarn es snapshot --ssl -E path.data=../data` in order to keep using the same data
* verify that:
  * an index template was created - `.kibana-event-log-8.0.0-template`
  * there is an initial index for 7.13 - `.kibana-event-log-8.0.0-000001`
  * the initial index is linked to the event log policy
* verify that the `7.13.*` indices are no longer linked to the event log ILM policy
  * can also do this via `https://localhost:9200/.kibana-event-log*/_ilm/explain`, where all the 7.13.3 indices should now show `managed: false`
* wait 10-ish minutes and verify that the `8.0.0` event log rolled over, but the `7.13.3` event logs did not

### Checklist

Delete any items that are not applicable to this PR.
